### PR TITLE
Add snackbar notification UI

### PR DESCRIPTION
**Pull request by marianamarasoiu**
Hi @lukechurch and @danrubel 

As discussed, this adds a notification via a snackbar at the bottom of the screen similar to the one from Coda. Let me know if you have any feedback on the design - happy to make any adjustments etc.

Screenshot:
Screenshot 2020-01-01 at 21 27 46

Thanks!

### CODE REVIEW

File: webapp/lib/view.dart [hunk 1]
```diff
@@ -1019,3 +1024,41 @@ class UrlView {
...
+    _contents.text = message;
+    snackbarElement.classes.remove('hidden');
+    snackbarElement.setAttribute('type', type.toString().replaceAll('SnackbarNotificationType.', ''));
+    new Timer(new Duration(seconds: 3), () => hideSnackbar());
+  }
+
+  hideSnackbar() {
+    snackbarElement.classes.add('hidden');
+    snackbarElement.attributes.remove('type');
+    // Remove the contents after the animation ends - the duration here must match the animation lenth
```
Comment (lukechurch) [inline webapp/lib/view.dart]: nit: s/lenth/length

Comment (lukechurch) [reply webapp/lib/view.dart]: Could you make it explicit that the thing it has to match is defined in snackbar.css?

File: webapp/lib/view.dart [hunk 2]
```diff
@@ -1019,3 +1024,41 @@ class UrlView {
...
+    _contents = new DivElement()
+      ..classes.add('contents');
+    snackbarElement.append(_contents);
+  }
+
+  showSnackbar(String message, SnackbarNotificationType type) {
+    _contents.text = message;
+    snackbarElement.classes.remove('hidden');
+    snackbarElement.setAttribute('type', type.toString().replaceAll('SnackbarNotificationType.', ''));
+    new Timer(new Duration(seconds: 3), () => hideSnackbar());
```
Comment (lukechurch) [inline webapp/lib/view.dart]: Could you a couple of these magic numbers into named constants defined on the class? E.g. I think this is the amount of time that the snackbar will be visible for?

File: webapp/lib/view.dart (hunk 1 above)
Comment (marianamarasoiu) [reply webapp/lib/view.dart]: Done, thanks!

File: webapp/lib/view.dart (hunk 2 above)
Comment (marianamarasoiu) [reply webapp/lib/view.dart]: Good idea, done now!

**Comment by marianamarasoiu** [comment]
Thanks both for the reviews, merging now!

**Pull request merged**